Agenda Maker/src/code.js

```diff
@@ -0,0 +1,146 @@
+  body.appendParagraph( "- Accountable: ");
+  body.appendParagraph( "- Consult: ");
+  body.appendParagraph( "- Inform: ");
+  body.appendParagraph( " ");
+  body.appendParagraph( "- Takeaway 2: ").editAsText().setBold(true);
+  body.appendParagraph( "- Responsible: ").editAsText().setBold(false);
+  body.appendParagraph( "- Accountable: ");
+  body.appendParagraph( "- Consult: ");
+  body.appendParagraph( "- Inform: ");
+  body.appendParagraph( " ");
+  body.appendParagraph( "- Takeaway 3: ").editAsText().setBold(true);
+  body.appendParagraph( "- Responsible: ").editAsText().setBold(false);
+  body.appendParagraph( "- Accountable: ");
+  body.appendParagraph( "- Consult: ");
+  body.appendParagraph( "- Inform: ");
+
+  folder.addFile(DriveApp.getFileById(doc.getId()));
+  
+
+  return doc.getId();
+
+}
+
+
+/*
+This is triggered whenever there is a change to the calendar
+When there is a change, it searches for events that include "#agenda" in the decription.
+*/
+function onCalendarChange(){
+  
+  //Get recent events with the tag
+  var now = new Date();
+  var events = CalendarApp.getEvents(now, new Date(now.getTime() + (2 * 60 * 60 * 1000000)),{search: '#agenda'});
+  
+  var folderId = checkFolder();
+  var templateId = getTemplateId(folderId);
+  
+  var folder = DriveApp.getFolderById(folderId); 
+  
+  //Go through any events found
+  for (i = 0; i < events.length; i++) {
+    var event = events[i];
+    
+    //Confirm whether the event has the tag
+    var description = event.getDescription();
+    if(description.search("#agenda")== -1) continue;
+
+    //Only work with events created by the owner of this calendar
+    if (event.isOwnedByMe()){
+
+      //Create a new document from the template, for an agenda for this event
+      var newDoc = DriveApp.getFileById(templateId).makeCopy();
+      newDoc.setName("Agenda for " + event.getTitle());
+
+      var file = DriveApp.getFileById(newDoc.getId());
+      folder.addFile(file);
+
+      var doc = DocumentApp.openById(newDoc.getId());
+      var body = doc.getBody();
+
+      //Fill in the template with information about the attendees from the calendar event
+      var conf = body.findText("##Attendees##");
+      if (conf){
+        var ref = conf.getStartOffset();
+
+        for (var i in event.getGuestList()){
+          var guest = event.getGuestList()[i];
+        
+          body.insertParagraph(ref + 2, guest.getEmail());
+        
+        }
+        body.replaceText("##Attendees##", "Attendees");
+      }
+      
+      //Replace the tag with a link to the agenda document
+      var agendaUrl = "https://docs.google.com/document/d/" + newDoc.getId();
+      description = description.replace("##agenda", "<a href=" + agendaUrl + ">Agenda Doc</a>");
+      event.setDescription(description);
+      
+      //Invite attendees to the Google doc so they automatically receive access to the agenda
+      newDoc.addEditor(newDoc.getOwner());
```

Comment:
does only add the owner? Could you save the attendees in the for loop above and add them as viewers or editors?